src/Servers/IIS/IIS/src/Core/IISHttpServer.cs

```diff
@@ -170,7 +170,7 @@ private static void OnDisconnect(IntPtr pvManagedHttpContext)
             try
             {
                 context = (IISHttpContext)GCHandle.FromIntPtr(pvManagedHttpContext).Target;
-                context.AbortIO(clientDisconnect: true);
+                context?.AbortIO(clientDisconnect: true);
```

Comment:
I doubt this fixes #17516. IISHttpContext.AbortIO() is on the stack posted in the issue, so context cannot be null.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContext.AbortIO(Boolean clientDisconnect)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpServer.OnDisconnect(IntPtr pvManagedHttpContext)
```

We think this fix is to initialize IISHttpContext._bodyOutput in the IISHttpContext constructor instead of the start of ProcessRequestsAsync. Maybe just calling InitializeContext() from the constructor is the way to do that. We just need to find the time to verify this won't break anything else.